python/erdos/streams.py

```diff
@@ -58,6 +59,147 @@ def name(self) -> str:
     def name(self, name: str):
         self._internal_stream.set_name(name)
 
+    def map(self, function: Callable[[Any], Any]) -> "OperatorStream":
+        """Applies the given function to each received value on the stream, and outputs
+        the results on the returned stream.
+
+        Args:
+            function (Callable[[Any], Any]): The function to be applied on each message
+                received on the input stream.
+
+        Returns:
+            An :py:class:`OperatorStream` that carries the results of the applied
+            function.
+        """
+
+        def map_fn(serialized_data: bytes) -> bytes:
+            result = function(pickle.loads(serialized_data))
+            return pickle.dumps(result)
+
+        return OperatorStream(self._internal_stream._map(map_fn))
+
+    def flat_map(self, function: Callable[[Any], Sequence[Any]]) -> "OperatorStream":
+        """Applies the given function to each received value on the stream, and outputs
+        the sequence of received outputs as individual messages.
+
+        Args:
+            function (Callable[[Any], Sequence[Any]]): The function to be applied to
+                each message received on the input stream.
+
+        Returns:
+            An :py:class:`OperatorStream` that carries the results of the applied
+            function.
+        """
+
+        def flat_map_fn(serialized_data: bytes) -> Sequence[bytes]:
+            mapped_values = function(pickle.loads(serialized_data))
+            result = []
+            for element in mapped_values:
+                result.append(pickle.dumps(element))
+            return result
+
+        return OperatorStream(self._internal_stream._flat_map(flat_map_fn))
+
+    def filter(self, function: Callable[[Any], bool]) -> "OperatorStream":
+        """Applies the given function to each received value on the stream, and outputs
+        the value if the function evaluates to true.
+
+        Args:
+            function (Callable[[Any], bool]): The function to be applied to each
+                message received on the input stream.
+
+        Returns:
+            An :py:class:`OperatorStream` that carries the filtered results from the
+            applied function.
+        """
+
+        def filter_fn(serialized_data: bytes) -> bool:
+            return function(pickle.loads(serialized_data))
+
+        return OperatorStream(self._internal_stream._filter(filter_fn))
+
+    def split(
+        self, function: Callable[[Any], bool]
+    ) -> Tuple["OperatorStream", "OperatorStream"]:
+        """Applies the given function to each received value on the stream, and outputs
+        the value to either the left or the right stream depending on the returned
+        boolean value from the function.
+
+        Args:
+            function (Callable[[Any], bool]): The function to be applied to each
+                message received on the input stream.
```

Comment:
Could you note which stream the boolean values correspond to? It's currently not clear if `True` corresponds to the left stream or the right stream.